pkg/filter/validator/basicauth.go

```diff
@@ -89,16 +92,26 @@ type (
 		authorizedUsersCache AuthorizedUsersCache
 	}
 
-	credentials struct {
+	// etcdCredentials defines the format for credentials in etcd
+	etcdCredentials struct {
 		Key      string `yaml:"key" jsonschema:"omitempty"`
-		Password string `yaml:"password" jsonschema:"omitempty"`
+		Username string `yaml:"username" jsonschema:"omitempty"`
+		Password string `yaml:"password" jsonschema:"required"`
 	}
 )
 
 const (
 	customDataPrefix = "/custom-data/"
 )
 
+// DefaultUsername uses Username if present, otherwise Key
+func (cred *etcdCredentials) DefaultUsername() string {
```

Comment:
propose to rename this function, `DefaultUsername` makes me think there's a default name.

Reply:
Good point, it's quite confusing name.. https://github.com/megaease/easegress/pull/472/commits/69ad26083b3492c55b9e669ab5738cc43c8831ab renames it to `Username()`